homeassistant/components/androidtv/config_flow.py

```diff
@@ -372,14 +373,9 @@ def _async_rules_form(
         )
 
 
-def _validate_state_det_rules(state_det_rules: str) -> list[Any] | None:
+def _validate_state_det_rules(state_det_rules: Any) -> list[Any] | None:
     """Validate a string that contain state detection rules and return a dict."""
-    try:
-        json_rules = json.loads(state_det_rules)
-    except ValueError:
-        _LOGGER.warning("Error loading state detection rules")
-        return None
-
```

Comment:
The object selector does not provide JSON validation, why was this removed? Isn't it needed anymore?

Reply:
This was used to convert the string from the text field to a `json` object. Because object selector already return a `json` object, conversion is not necessary and validation is done by  `state_detection_rules_validator` method.